source/slang/hlsl.meta.slang

```diff
@@ -1654,15 +1654,44 @@ extension __TextureImpl<T,Shape,isArray,isMS,sampleCount,0,isShadow,0,format>
         }
     }
 
+    [__readNone]
+    [ForceInline]
+    void __checkSupportedType()
+    {
+        __target_switch
+        {
+        case glsl:
+            static_assert(!(T is double)
+                && !(T is vector<double,1>)
+                && !(T is vector<double,2>)
+                && !(T is vector<double,3>)
+                && !(T is vector<double,4>),
+                "GLSL doesn't support texture sampling for double types.");
+            return;
+        case hlsl:
+            static_assert(T is IFloat, "HLSL doesn't support texture sampling for integer types.");
+            static_assert(!(T is double)
+                && !(T is vector<double,1>)
+                && !(T is vector<double,2>)
+                && !(T is vector<double,3>)
+                && !(T is vector<double,4>),
+                "HLSL doesn't support texture sampling for double types.");
+            return;
+        default: return;
+        }
+    }
+
     [__readNone]
     [ForceInline]
     [require(cpp_cuda_glsl_hlsl_metal_spirv, texture_sm_4_1)]
     T SampleLevel(SamplerState s, vector<float, Shape.dimensions+isArray> location, float level)
     {
+        __checkSupportedType();
         __target_switch
         {
             case cpp:
             case hlsl:
+                static_assert(T is IFloat, "HLSL doesn't support texture sampling for integer types.");
```

Comment:
`static_assert` seems to be working as expected if it is used directly inside of `__target_switch`.
That being said, the following line works as expected,
```
static_assert(T is IFloat, "HLSL doesn't support texture sampling for integer types.");
```
But if I call it indirectly inside of another function it doesn't work.
In this case the function call to `__checkSupportedType()` itself doesn't exist in the IR-dump.
I think there is a bug when `__intrinsic_asm` is used.

Here is IR dump for more detail,
```
generic %150    : %144
{
block %151(
                [TypeConstraintDecoration(%IFloat)]
                [nameHint("T")]
                param %T1       : type_t,
                [TypeConstraintDecoration(%x5Fx5FITextureShape)]
                [nameHint("Shape")]
                param %Shape1   : type_t,
                [nameHint("isArray")]
                param %isArray1 : Int,
                [nameHint("isMS")]
                param %isMS1    : Int,
                [nameHint("sampleCount")]
                param %sampleCount1     : Int,
                [nameHint("isShadow")]
                param %isShadow1        : Int,
                [nameHint("format")]
                param %format1  : Int,
                param %152      : witness_table_t(%IFloat),
                param %153      : witness_table_t(%x5Fx5FITextureShape)):
        let  %154       : Int   = lookupWitness(%153, %58)
        let  %155       : Int   = lookupWitness(%153, %139)
        let  %156       : Int   = add(%isArray1, %154)
        [ForceInline]
        [readNone]
        [nameHint("SampleLevel")]
                func %SampleLevel       : Func(%T1, TextureType(%T1, %Shape1, %isArray1, %isMS1, %sampleCount1, 0 : Int, %isShadow1, 0 : Int, %format1, %153), SamplerState, Vec(Float, %156), Float)
        {
block %157(
                        [nameHint("this")]
                        param %this     : TextureType(%T1, %Shape1, %isArray1, %isMS1, %sampleCount1, 0 : Int, %isShadow1, 0 : Int, %format1, %153),
                        [nameHint("s")]
                        param %s        : SamplerState,
                        [nameHint("location")]
                        param %location : Vec(Float, %156),
                        [nameHint("level")]
                        param %level    : Float):
                unconditionalBranch(%158)

block %158:
                StaticAssert(true, "HLSL doesn\'t support texture sampling for integer types.")
                GenericAsm(".SampleLevel")
}
        return_val(%SampleLevel)
}
```
I expect that the function call to `__checkSupportedType()` should appear above the line, `unconditionalBranch(%158)`.
But it is not there.

Ariel and I have seen a problem like this before.
I will debug and see what is happening.

I am guessing that it could be by design that when we call `__intrinsic_asm`, we may take a short-cut to skip functions calls.